src/Util/PhpServicesCreator.php

```diff
@@ -0,0 +1,610 @@
+                    ],
+                    'stmts' => $closureStmt,
+                ])
+            )
+        );
+
+        $node = $this->creatorFactory->getNode();
+        $beforeClosure = !empty($this->useStatements) ? "\n" : null;
+        // add a blank line between the last use statement and the closure
+        array_splice($node->stmts,
+            -1,
+            0,
+            [new Node\Name($beforeClosure."/*\n * This file is the entry point to configure your own services.\n */")]
+        );
+
+        return (new Standard())->prettyPrintFile([$node])."\n";
+    }
+
+    /**
+     * @return Node[]
+     */
+    private function createClosureStmts(array $yamlData): array
+    {
+        foreach ($yamlData as $key => $values) {
+            if (null === $values) {
+                // if the key is written without values, we declare the variable ($parameters/$services).
+                $values = [];
+            }
+            switch ($key) {
+                case 'parameters':
+                    $this->addParametersNodes($values);
+                    break;
+                case 'imports':
+                    $this->addImportsNodes($values);
+                    break;
+                case 'services':
+                    $this->addServicesNodes($values);
+                    break;
+                default:
+                    throw new \LogicException(sprintf('The key %d is not available', $key));
+                    break;
+            }
+        }
+
+        sort($this->useStatements);
+        // Add at the end to sort them in alphabetical order.
+        foreach ($this->useStatements as $className) {
+            $this->creatorFactory->addStmt(
+                $this->factory->use($className)
+            );
+        }
+
+        // remove the last carriage return "\n" if exists.
+        $this->stmts[array_key_last($this->stmts)]->parts[0] = rtrim($this->stmts[array_key_last($this->stmts)]->parts[0], "\n");
+
+        return $this->stmts;
+    }
+
+    private function addParametersNodes(array $parameters)
+    {
+        $this->addLineStmt('// Put parameters here that don\'t need to change on each machine where the app is deployed.');
+        $this->addLineStmt('// https://symfony.com/doc/current/best_practices/configuration.html#application-related-configuration.');
+
+        $this->addLineStmt('$parameters = $configurator->parameters();', true);
+
+        foreach ($parameters as $parameterName => $value) {
+            $value = $this->toString($value, true);
+            $this->addLineStmt(sprintf(
+                '$parameters->set(%s, %s);',
+                $this->createStringArgument($parameterName),
+                $value
+            ), true);
+        }
+    }
+
+    private function addImportsNodes(array $imports)
+    {
+        foreach ($imports as $import) {
+            $import = \is_string($import) ? $this->createStringArgument($import) : $this->createListFromArray($import);
```

Comment:
There are 2 cases here:

```yml
imports:
    # case 1
    - other_file.yaml
    # case 2
    - { resource: 'somefile.yaml', 'type': 'annotations', ignore_errors: true }
```

Your handling of case 1 makes sense to me - 👍 

But I don't think case 2 is correct. If I understand things correctly, `$this->createListFromArray($import)` just lists the arguments by *order* in YAML. So:

```yml
imports:
    # will become $configurator->import('somefile.yaml', 'annotations', true);
    # PERFECT!
    - { resource: 'somefile.yaml', 'type': 'annotations', ignore_errors: true }

    # will become $configurator->import(true, 'somefile.yaml', 'annotations');
    # WRONG!
    - { ignore_errors: true, resource: 'somefile.yaml', type: 'annotations' }
```

Instead, if `$import` is an array, we actually need to read the keys and figure out the values:

```php
$type = null;
$ignoreErrors = false;
if (is_array($import)) {
    $resource = $import['resource'];
    $type = $import['type'] ?? null;
    $ignoreErrors = $import['ignore_errors'] ?? false;
} else {
    $resource = $import;
}

// build the `$configurator->import()` based on this information.
```